eng/Directory.Build.Common.props

```diff
@@ -127,6 +127,8 @@
     <RequiredTargetFrameworks>net7.0;net6.0</RequiredTargetFrameworks>
     <!-- Also test net461 on Windows. -->
     <RequiredTargetFrameworks Condition="'$(OS)' == 'Windows_NT'">$(RequiredTargetFrameworks);net461</RequiredTargetFrameworks>
+    <!-- But only build snippets for the latest. -->
+    <RequiredTargetFrameworks Condition="'$(BuildSnippets)' == 'true'">net7.0</RequiredTargetFrameworks>
```

Comment:
With this change we definitely need to ensure we always keep BuildSnippets separate from the normal build and package commands. I somewhat fear we might end up dropping target frameworks for what we ship if someone starts passing this property to our normal builds.